Ical.Net.Tests/WikiSamples/RecurrenceWiki.md

```diff
@@ -0,0 +1,414 @@
+    Interval = 2,
+    Count = 30
+    // Add other parameters like ByDay, ByMonth, etc.
+};
+
+var calendarEvent = new CalendarEvent()
+{
+    DtStart = new CalDateTime(2025, 07, 10),
+    DtEnd = new CalDateTime(2025, 07, 11),
+    // Add the rule to the event.
+    RecurrenceRules = [recurrence]
+};
+```
+### How to get occurrences of a recurring event
+After creating a recurring event we can call `GetOccurrences()` to get all occurrences of a series or `GetOccurrences(CalDateTime startTime)` to get all occurrences after a given date. We can limit the evaluation by adding a `TakeWhileBefore(CalDateTime periodEnd)` to get occurrences till a specific point in time.
+
+> [!NOTE]
+> The provided end-date in the `TakeWhileBefore` method calculate against the event start is meant in an exclusive manner. In our example, the occurrence on 2025-08-01 is not provided despite the event-start (2025-08-01) and the calculation-end (2025-08-01) is the same date (exclusive). To include it we have to set the calculation-end to 2025-08-01T00:00:01.
+
+> [!WARNING]
+> Calculating a series with no end may lead in a `EvaluationOutOfRangeException`. Make sure you use `TakeWhileBefore` to avoid this.
+
+```cs
+// Get all occurrences of the series.
+IEnumerable<Occurrence> allOccurrences = calendarEvent.GetOccurrences();
+Assert.That(allOccurrences.Count(), Is.EqualTo(30));
+
+// Get the occurrences in July.
+IEnumerable<Occurrence> julyOccurrences = calendarEvent
+    .GetOccurrences(new CalDateTime(2025, 07, 01))
+    .TakeWhileBefore(new CalDateTime(2025, 08, 01));
+Assert.That(julyOccurrences.Count(), Is.EqualTo(11));
+```
+
+> [!TIP]
+> We can call the `GetOccurrences` method on `CalendarEvent` and on `Calendar`.
+
+
+## Recurrence Examples
+### Daily, Interval, Count
+Suppose we want to create an event for July 10, between 09:00 and 10:00 that recurs every other day for 2 occurrences.
+
+```cs
+// Create the CalendarEvent
+var start = new CalDateTime(2025, 07, 10, 09, 00, 00, "Europe/Zurich");
+var recurrence = new RecurrencePattern()
+{
+	Frequency = FrequencyType.Daily,
+	Interval = 2,
+	Count = 2
+};
+
+var calendarEvent = new CalendarEvent()
+{
+	DtStart = start,
+	DtEnd = start.AddHours(1),
+	RecurrenceRules = [recurrence]
+};
+
+// Add CalendarEvent to Calendar
+var calendar = new Calendar();
+calendar.Events.Add(calendarEvent);
+
+// Serialize Calendar to string
+var calendarSerializer = new CalendarSerializer();
+var calendarAsIcs = calendarSerializer.SerializeToString(calendar);
+```
+This results in the following Ics string.
+```ics
+BEGIN:VCALENDAR
+BEGIN:VEVENT
+DTEND;TZID=Europe/Zurich:20250710T100000
+DTSTART;TZID=Europe/Zurich:20250710T090000
+RRULE:FREQ=DAILY;INTERVAL=2;COUNT=2
+END:VEVENT
+END:VCALENDAR
+```
+Now let's get all occurrences of this series.
+```cs
+var occurrences = calendar.GetOccurrences();
+```
+This will give us two occurrences with the following start and end dates.
+```ics
+DTEND;TZID=Europe/Zurich:20250710T100000
```

Comment:
We should not use `ICS` notation for occurrences. Let's show the values of that are returned.
Maybe have a look this example: https://github.com/ical-org/ical.net/blob/bd4225eb4670d0d3f212daf57a0bc81db13d7279/Ical.Net.Tests/Logging.Tests/TestLoggingManagerTests.cs#L18
It's just a demo how we can use the new logger, so feel free to modify.

Reply:
Anyway, the logger may be a good tool create the results from the Wiki samples.

Reply:
This was just meant to be a sample for how ro use the logger - don't use 1:1 for the Wiki